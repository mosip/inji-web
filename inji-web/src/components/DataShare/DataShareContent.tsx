import React, {useState} from "react";
import {MdOutlineKeyboardArrowDown} from "react-icons/md";
import {DataShareDisclaimer} from "./DataShareDisclaimer";
import {useDispatch, useSelector} from "react-redux";
import {RootState} from "../../types/redux";
import {storevcStorageExpiryLimitInTimes} from "../../redux/reducers/commonReducer";
import {useTranslation} from "react-i18next";
import {DSContentProps} from "../../types/components";
import { toast } from "react-toastify";

export const DataShareContent:React.FC<DSContentProps> = (props) => {

    const [timesDropDown, setTimesDropDown] = useState<boolean>(false);
    const vcStorageExpiryLimitInTimes = useSelector((state: RootState) => state.common.vcStorageExpiryLimitInTimes);
    const dispatch = useDispatch();
    const {t} = useTranslation("DataShareExpiryModal");

    const getExpiryDisplayName = (expiry: number) => {
        let expiryDisplayName = expiry.toString();
        switch(expiry){
            case 1:
                expiryDisplayName = t("content.validityTimesOptions.once");
                break;
            case 3:
                expiryDisplayName = t("content.validityTimesOptions.thrice");
                break;
            case -1:
                expiryDisplayName = t("content.validityTimesOptions.noLimit");
                break;
        }
        return expiryDisplayName;
    }


    return <div className="relative px-6 flex flex-col justify-between border-b border-solid border-iw-borderLight" data-testid={"DataShareContent-Outer-Container"}>
        <div className="relative flex border-b-2">
            <div className="my-4 text-iw-subTitle text-lg w-1/3 leading-relaxed py-5" data-testid={"DataShareContent-Outer-Title"}>{t("content.selectedDocument")}</div>
            <div className={"my-4 text-center shadow-md rounded-lg w-2/3 flex flex-row py-5 items-center"}>
                <div className={"px-6"} data-testid={"DataShareContent-Issuer-Logo-Container"}>
                    <img data-testid={"DataShareContent-Issuer-Logo"} src={props.credentialLogo} alt="Issuer Logo"
                         className="w-12 h-12 flex"/>
                </div>
                <div className="text-center text-lg" data-testid={"DataShareContent-Issuer-Name"}>{props.credentialName}</div>
            </div>
        </div>
        <div className="relative flex">
            <div className="my-4 w-1/3 text-iw-subTitle text-lg leading-relaxed py-5" data-testid={"DataShareContent-Consent-Container"}>{t("content.consent")}</div>
            <div className={"my-4 w-2/3 flex flex-row py-5 items-center"}>
                <label className="w-1/2 flex items-center">
                    <input
                        type="radio"
                        name="consentValidity"
                        value="number"
                        checked={true}
                        data-testid={"DataShareContent-Consent-Radio"}
                        className="accent-iw-primary scale-150 mx-2"
                    />
                    <span className="font-base text-iw-subTitle" data-testid={"DataShareContent-Consent-Option"}>{t("content.validityTimesHeader")}</span>
                </label>

                <label onClick={()=>toast.warning(t("toastText"))} className="w-1/2 flex items-center" data-testid={"DataShareContent-Validity-Date"}>  
                    <div className="relative inline-block">
                        <input 
                            type="radio" 
                            name="consentValidity" 
                            value="date" 
                            disabled={true} 
                            className="accent-iw-primary scale-150 mx-2" 
                        />
                        <div 
                            className="absolute inset-0"
                            onClick={(e) => {e.stopPropagation(); toast.warning(t("toastText")); }}
                        />
                    </div>
                        <span className="font-light text-iw-subTitle" data-testid={"DataShareContent-Validity-Date-Title"}>
                        {t("content.validityDate")}</span>
                </label>
            </div>
        </div>
        <div className="relative flex mb-4" data-testid={"times-dropdown"} onClick={()=>setTimesDropDown(times => !times)}>
                    <div className={"w-1/3"}></div>
                    <div className={"w-2/3 py-4 px-4 rounded-lg border-2 border-iw-borderLight flex flex-row items-center"}>
                        <label className={"w-full h-full"} data-testid={"DataShareContent-Selected-Validity-Times"}>{getExpiryDisplayName(vcStorageExpiryLimitInTimes)}</label>
                        <MdOutlineKeyboardArrowDown size={30} color={'var(--iw-color-arrowDown)'} />
                    </div>
                </div>

                {timesDropDown && <div className="relative flex mb-4">
                    <div className={"w-1/3"}></div>
                    <div
                        className={"w-2/3 py-4 px-2 border-2 border-iw-borderLight rounded-lg shadow-lg shadow-iw-shadow flex flex-col justify-center items-center"} data-testid={"DataShareContent-Validity-Times-DropDown"}>
                        <label data-testid={"DataShareContent-Validity-Times-DropDown-Once"} onClick={()=>{dispatch(storevcStorageExpiryLimitInTimes(1)); setTimesDropDown(false)} } className={"w-full h-full py-3 px-4 hover:bg-iw-borderLight hover:rounded-lg"}>{t("content.validityTimesOptions.once")}</label>
                        <label data-testid={"DataShareContent-Validity-Times-DropDown-Thrice"} onClick={()=>{dispatch(storevcStorageExpiryLimitInTimes(3)); setTimesDropDown(false)} } className={"w-full h-full py-3 px-4 hover:bg-iw-borderLight hover:rounded-lg"}>{t("content.validityTimesOptions.thrice")}</label>
                        <label data-testid={"DataShareContent-Validity-Times-DropDown-NoLimit"} onClick={()=>{dispatch(storevcStorageExpiryLimitInTimes(-1)); setTimesDropDown(false)} } className={"w-full h-full py-3 px-4 hover:bg-iw-borderLight hover:rounded-lg"}>{t("content.validityTimesOptions.noLimit")}</label>
                        <label data-testid={"DataShareContent-Validity-Times-DropDown-Custom"} onClick={()=>{setTimesDropDown(false); props.setIsCustomExpiryInTimesModalOpen(true)} }
                               className={"w-full h-full py-3 px-4 hover:bg-iw-borderLight hover:rounded-lg"}>{t("content.validityTimesOptions.custom")}</label>
                    </div>
                </div>}
        <DataShareDisclaimer content={t("disclaimer")}/>
    </div>;
}

