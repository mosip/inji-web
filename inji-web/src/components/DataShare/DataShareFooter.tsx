import React from "react";
import {DSFooterProps} from "../../types/components";
import {BorderedButton} from "../Common/Buttons/BorderedButton";
import {SolidButton} from "../Common/Buttons/SolidButton";

export const DataShareFooter: React.FC<DSFooterProps> = (props) => {
    return (
        <div
            className="flex items-center justify-between p-6 rounded-b"
            data-testid={`DataShareFooter-Outer-Container${
                props.parent ? `-${props.parent}` : ""
            }`}
        >
            <BorderedButton
                testId={"DataShareFooter-Cancel-Button"}
                onClick={props.onCancel}
                title={props.cancelText}
                fullWidth={true}
            />
            <SolidButton
                testId={"DataShareFooter-Success-Button"}
                onClick={props.onSuccess}
                title={props.successText}
                fullWidth={true}
            />
        </div>
    );
};
