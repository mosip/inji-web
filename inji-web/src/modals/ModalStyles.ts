export const ModalStyles = {
    confirmation: {
        container: "flex flex-col items-center pt-4 pb-4 px-8 gap-3",
        title: "text-2xl justify-center font-medium text-center text-[--iw-color-textTertiary]",
        message: "text-[--iw-color-textTertiary] font-base font-light text-sm",
        buttonsContainer: "flex items-center justify-around sm:flex-row flex-col gap-4 w-full pt-3",
        cancelButton: "py-2"
    },
    modal: {
        overlay: "fixed inset-0 z-50 flex items-center justify-center bg-black bg-opacity-50",
        container: "bg-white rounded-lg shadow-lg mx-4 my-2 mt-10 flex flex-col relative pt-6 border-4 border-white",
        header: {
            wrapper: "mb-4 flex items-center justify-between px-6 flex-shrink-0 gap-4",
            closeButton: "text-gray-500 hover:text-gray-700 text-2xl font-bold",
        },
        separator: "flex-shrink-0",
        content: {
            wrapper: "flex flex-col flex-1 relative sm:bg-transparent bg-white sm:mb-4 pb-0 min-h-0",
            container: "overflow-y-auto flex-1 min-h-0 sm:mx-5 sm:m-0 m-2 sm:bg-transparent bg-white sm:rounded-none rounded-xl"
        },
        action: "sm:absolute sm:right-6 sm:bottom-0 sm:mr-10 sm:pb-8 sm:bg-transparent sm:w-auto static bg-white flex px-2 w-full py-2 sm:rounded-b-lg"
    }
}