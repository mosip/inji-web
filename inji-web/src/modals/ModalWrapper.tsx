import React,{useEffect} from 'react';
import {ModalWrapperProps} from "../types/components";
export const ModalWrapper: React.FC<ModalWrapperProps> = (props) => {

    // Effect to lock/unlock scroll on body when modal is open/closed
    useEffect(() => {
      // Lock scrolling when the modal is open
      document.body.style.overflow = 'hidden';
  
      // Cleanup: Unlock scrolling when the modal is closed
      return () => {
        document.body.style.overflow = '';
      };
    }, []); // Empty dependency array ensures this runs on mount/unmount only

  return <>
   <div data-testid="ModalWrapper-Outer-Container" className={`fixed inset-0 ${props.zIndex == 50 ? 'z-50' : 'z-40'} overflow-y-auto overflow-x-hidden`}>
      <div className="min-h-full p-0 flex items-center justify-center">
        <div className={`w-auto my-8 mx-4 sm:mx-6 ${props.size === '3xl' ? 'max-w-3xl' : 'max-w-sm'} min-w-[320px]`}>
          <div data-testid="ModalWrapper-Inner-Container" className="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-iw-background outline-none focus:outline-none">
            {props.header}
            {props.content}
            {props.footer}
          </div>
        </div>
      </div>
    </div>

    <div data-testid="ModalWrapper-BackDrop" className={`opacity-25 fixed inset-0 ${props.zIndex == 50 ? 'z-40' : 'z-30'} bg-iw-backDrop`}></div>
  </>;
};