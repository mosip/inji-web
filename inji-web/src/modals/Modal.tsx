import React, {Fragment} from 'react';
import ReactDOM from 'react-dom';
import {BackArrowButton} from "../components/Common/Buttons/BackArrowButton";
import {PageTitle} from "../components/Common/PageTitle/PageTitle";
import {Clickable} from "../components/Common/Clickable";
import {Separator} from "../components/Common/Separator";
import {CloseIconButton} from "../components/Common/Buttons/CloseIconButton";
import {ModalStyles} from "./ModalStyles";

interface ModalProps {
    isOpen: boolean;
    onClose: () => void;
    title?: string;
    children: React.ReactNode;
    action?: React.ReactNode;
    size?: "sm" | "md" | "lg" | "xl" | "2xl" | "3xl";
    testId: string;
}

/**
 * Modal component that renders a dialog box with a backdrop.
 * Features:
 * 1. Renders provided title with back arrow and on close as header
 *    Renders the provided children
 *    If action is provided
 *      - action is sticky floatable for large screens
 *      - container block for smaller screens
 * 2. Serves a responsive modal which can be used to wrap any content as per need
 */
export const Modal: React.FC<ModalProps> = ({isOpen, onClose, children, action, title, size, testId}) => {
    if (!isOpen) return null;

    const modalSize = size ? `min-h-${size}` : "h-[83vh] w-[90vw] sm:w-[70vw] sm:h-[80vh]"

    return ReactDOM.createPortal(
        <Clickable
            className={ModalStyles.modal.overlay}
            onClick={onClose}
            testId={`${testId}-modal-overlay`}
        >
            <dialog
                className={`${ModalStyles.modal.container} ${modalSize}`}
                aria-modal="true"
                open
                onClose={onClose}
                onCancel={onClose}
            >
                {
                    title && (
                        <Fragment>
                            <div className={ModalStyles.modal.header.wrapper}>
                                <BackArrowButton onClick={onClose}/>
                                <PageTitle value={title} testId={"title-modal"}/>
                                <CloseIconButton
                                    onClick={onClose}
                                    btnClassName={ModalStyles.modal.header.closeButton}
                                />
                            </div>
                            <Separator className={ModalStyles.modal.separator}/>
                        </Fragment>
                    )
                }

                <div className={ModalStyles.modal.content.wrapper}>
                    <div className={ModalStyles.modal.content.container}>
                        {children}
                    </div>

                    {action && (
                        <div className={ModalStyles.modal.action}>
                            {action}
                        </div>
                    )}
                </div>
            </dialog>
        </Clickable>,
        document.body
    );
};