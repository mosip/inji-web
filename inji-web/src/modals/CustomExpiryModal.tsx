import React, {useState} from "react";
import {ModalWrapper} from "./ModalWrapper";
import {DataShareFooter} from "../components/DataShare/DataShareFooter";
import {CustomExpiryTimesContent} from "../components/DataShare/CustomExpiryTimes/CustomExpiryTimesContent";
import {CustomExpiryTimesHeader} from "../components/DataShare/CustomExpiryTimes/CustomExpiryTimesHeader";
import {useDispatch, useSelector} from "react-redux";
import {storevcStorageExpiryLimitInTimes} from "../redux/reducers/commonReducer";
import {RootState} from "../types/redux";
import {useTranslation} from "react-i18next";
import {CustomExpiryModalProps} from "../types/components";
import {toast} from "react-toastify";

export const CustomExpiryModal: React.FC<CustomExpiryModalProps> = (props) => {
    const vcStorageExpiryLimitInTimes = useSelector(
        (state: RootState) => state.common.vcStorageExpiryLimitInTimes
    );
    const [expiryTime, setExpiryTime] = useState<number>(
        vcStorageExpiryLimitInTimes > 1 ? vcStorageExpiryLimitInTimes : 1
    );
    const {t} = useTranslation("CustomExpiryModal");
    const dispatch = useDispatch();

    const onSuccess = () => {
        if (expiryTime < 1 || expiryTime > 50) {
            toast.error(t("error"));
        } else {
            dispatch(storevcStorageExpiryLimitInTimes(expiryTime));
            props.onSuccess();
        }
    };

    return (
        <ModalWrapper
            header={<CustomExpiryTimesHeader title={t("title")} />}
            content={
                <CustomExpiryTimesContent
                    expiryTime={expiryTime}
                    setExpiryTime={setExpiryTime}
                />
            }
            footer={
                <DataShareFooter
                    parent="custom-expiry-modal"
                    successText={t("success")}
                    cancelText={t("cancel")}
                    onSuccess={onSuccess}
                    onCancel={props.onCancel}
                />
            }
            size={"sm"}
            zIndex={50}
        />
    );
};
