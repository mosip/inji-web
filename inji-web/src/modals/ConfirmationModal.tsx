import React from "react";
import {Modal} from "./Modal";
import {SolidButton} from "../components/Common/Buttons/SolidButton";
import {BorderedButton} from "../components/Common/Buttons/BorderedButton";
import {ModalStyles} from "./ModalStyles";
import {useTranslation} from "react-i18next";

export const ConfirmationModal = (props: {
    title: string;
    message: string;
    onConfirm: () => void;
    onCancel: () => void;
    testId: string;
}) => {
    const {t} = useTranslation('Common')

    return (
        <Modal
            isOpen={true}
            onClose={props.onCancel}
            size="md"
            testId={`${props.testId}-confirmation`}
        >
            <div className={ModalStyles.confirmation.container}>
                            <span data-testid={`title-${props.testId}`} className={ModalStyles.confirmation.title}>
                                {props.title}
                            </span>
                <div className={ModalStyles.confirmation.message}>
                    <p>{props.message}</p>
                </div>
                <div className={ModalStyles.confirmation.buttonsContainer}>
                    <BorderedButton
                        testId={"btn-cancel"}
                        onClick={props.onCancel}
                        title={t('cancel')}
                        fullWidth
                        className={ModalStyles.confirmation.cancelButton}
                    />
                    <SolidButton
                        testId={"btn-confirm"}
                        onClick={props.onConfirm}
                        title={t('confirm')}
                        fullWidth
                    />
                </div>
            </div>
        </Modal>
    );
}