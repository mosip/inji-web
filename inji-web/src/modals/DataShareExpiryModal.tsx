import React, {useState} from "react";
import {ModalWrapper} from "./ModalWrapper";
import {DataShareHeader} from "../components/DataShare/DataShareHeader";
import {DataShareFooter} from "../components/DataShare/DataShareFooter";
import {DataShareContent} from "../components/DataShare/DataShareContent";
import {CustomExpiryModal} from "./CustomExpiryModal";
import {useTranslation} from "react-i18next";
import {DataShareExpiryModalProps} from "../types/components";

export const DataShareExpiryModal: React.FC<DataShareExpiryModalProps> = (props) => {

    const [isCustomExpiryInTimesModalOpen, setIsCustomExpiryInTimesModalOpen] = useState<boolean>(false);
    const {t} = useTranslation("DataShareExpiryModal")
    return <React.Fragment>
            <ModalWrapper header={<DataShareHeader title={t("title")} subTitle={t("subTitle", {credentialName: props.credentialName})}/>}
                       content={<DataShareContent credentialName={props.credentialName} credentialLogo={props.credentialLogo} setIsCustomExpiryInTimesModalOpen={setIsCustomExpiryInTimesModalOpen}/>}
                       footer={<DataShareFooter parent = "data-share-expiry-modal" cancelText={t("cancel")} successText={t("success")} onSuccess={props.onSuccess} onCancel={props.onCancel}/>}
                       size={"3xl"}
                       zIndex={40}/>
            {isCustomExpiryInTimesModalOpen && <CustomExpiryModal onSuccess={() => setIsCustomExpiryInTimesModalOpen(false)} onCancel={() => setIsCustomExpiryInTimesModalOpen(false)}/>}
        </React.Fragment>

}

