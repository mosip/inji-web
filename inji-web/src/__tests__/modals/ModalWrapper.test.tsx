import {render, screen} from "@testing-library/react";
import {ModalWrapper} from "../../modals/ModalWrapper";
import {DataShareHeader} from "../../components/DataShare/DataShareHeader";
import {DataShareFooter} from "../../components/DataShare/DataShareFooter";
import React from "react";
import {DataShareContent} from "../../components/DataShare/DataShareContent";
import {reduxStore} from "../../redux/reduxStore";
import {Provider} from "react-redux";

describe("Test the Layout of the Modal Wrapper", () => {

    const customMockFn = jest.fn();
    beforeEach(() => {
        render(
            <Provider store={reduxStore}>
            <ModalWrapper header={<DataShareHeader title={"title"} subTitle={"subTitle"}/>}
                             content={<DataShareContent credentialName={"credentialName"} credentialLogo={"credentialLogo"} setIsCustomExpiryInTimesModalOpen={jest.fn()}/>}
                             footer={<DataShareFooter cancelText={"cancel"} successText={"success"} onSuccess={jest.fn()} onCancel={jest.fn()}/>}
                             size={"3xl"}
                             zIndex={40} />
            </Provider>)
    })

    test("Test the presence of the Outer Container", ()=>{
        const document = screen.getByTestId("ModalWrapper-Outer-Container");
        expect(document).toBeInTheDocument();
    })
    test("Test the presence of the Inner Container", ()=>{
        const document = screen.getByTestId("ModalWrapper-Inner-Container");
        expect(document).toBeInTheDocument();
        expect(document.children.length).toBe(3)
    })
    test("Test the presence of the Back Drop", ()=>{
        const document = screen.getByTestId("ModalWrapper-BackDrop");
        expect(document).toBeInTheDocument();
    })
    test("Snapshot test for ModalWrapper styling", () => {
        const { asFragment } = render(
            <Provider store={reduxStore}>
                <ModalWrapper 
                    header={<DataShareHeader title={"title"} subTitle={"subTitle"}/>}
                    content={<DataShareContent 
                        credentialName={"credentialName"} 
                        credentialLogo={"credentialLogo"} 
                        setIsCustomExpiryInTimesModalOpen={jest.fn()}
                    />}
                    footer={<DataShareFooter 
                        cancelText={"cancel"} 
                        successText={"success"} 
                        onSuccess={jest.fn()} 
                        onCancel={jest.fn()}
                    />}
                    size={"3xl"}
                    zIndex={40} 
                />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
})
