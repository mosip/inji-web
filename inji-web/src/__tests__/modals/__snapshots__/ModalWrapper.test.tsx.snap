// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Layout of the Modal Wrapper Snapshot test for ModalWrapper styling 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-40 overflow-y-auto overflow-x-hidden"
    data-testid="ModalWrapper-Outer-Container"
  >
    <div
      class="min-h-full p-0 flex items-center justify-center"
    >
      <div
        class="w-auto my-8 mx-4 sm:mx-6 max-w-3xl min-w-[320px]"
      >
        <div
          class="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-iw-background outline-none focus:outline-none"
          data-testid="ModalWrapper-Inner-Container"
        >
          <div
            class="flex items-start justify-between p-5 border-b border-solid rounded-t"
            data-testid="DataShareHeader-Outer-Container"
          >
            <div
              class="flex flex-col"
            >
              <div
                class="text-lg font-semibold"
                data-testid="DataShareHeader-Header-Title"
              >
                 title
              </div>
              <div
                class="text-md text-iw-subTitle"
                data-testid="DataShareHeader-Header-SubTitle"
              >
                 subTitle 
              </div>
            </div>
          </div>
          <div
            class="relative px-6 flex flex-col justify-between border-b border-solid border-iw-borderLight"
            data-testid="DataShareContent-Outer-Container"
          >
            <div
              class="relative flex border-b-2 space-x-2"
            >
              <div
                class="my-4 break-words text-iw-subTitle text-lg w-1/3 leading-relaxed py-5"
                data-testid="DataShareContent-Outer-Title"
              >
                Selected Document
              </div>
              <div
                class="my-4 text-center shadow-md rounded-lg w-2/3 flex flex-row py-5 items-center"
              >
                <div
                  class="px-4"
                  data-testid="DataShareContent-Issuer-Logo-Container"
                >
                  <img
                    alt="Issuer Logo"
                    class="w-12 h-12 min-w-[36px]"
                    data-testid="DataShareContent-Issuer-Logo"
                    src="credentialLogo"
                  />
                </div>
                <div
                  class="text-center text-lg w-full sm:w-auto break-words min-w-[100px]"
                  data-testid="DataShareContent-Issuer-Name"
                >
                  credentialName
                </div>
              </div>
            </div>
            <div
              class="relative flex space-x-10"
            >
              <div
                class="my-4 w-1/3 text-iw-subTitle text-lg leading-relaxed py-5"
                data-testid="DataShareContent-Consent-Container"
              >
                Consent Validity *
              </div>
              <div
                class="my-4 w-2/3 flex flex-row py-5 items-center"
              >
                <label
                  class="w-1/2 flex items-center"
                >
                  <input
                    checked=""
                    class="accent-iw-primary scale-150 mx-2"
                    data-testid="DataShareContent-Consent-Radio"
                    name="consentValidity"
                    type="radio"
                    value="number"
                  />
                  <span
                    class="font-base text-iw-subTitle"
                    data-testid="DataShareContent-Consent-Option"
                  >
                    Number of times
                  </span>
                </label>
                <label
                  class="w-1/2 flex items-center"
                  data-testid="DataShareContent-Validity-Date"
                >
                  <div
                    class="relative inline-block"
                  >
                    <input
                      class="accent-iw-primary scale-150 mx-2"
                      disabled=""
                      name="consentValidity"
                      type="radio"
                      value="date"
                    />
                    <div
                      class="absolute inset-0"
                    />
                  </div>
                  <span
                    class="font-light text-iw-subTitle"
                    data-testid="DataShareContent-Validity-Date-Title"
                  >
                    Date
                  </span>
                </label>
              </div>
            </div>
            <div
              class="relative flex mb-4"
              data-testid="times-dropdown"
            >
              <div
                class="w-1/3"
              />
              <div
                class="w-2/3 py-4 px-4 rounded-lg border-2 border-iw-borderLight flex flex-row items-center"
              >
                <label
                  class="w-full h-full"
                  data-testid="DataShareContent-Selected-Validity-Times"
                >
                  Once
                </label>
                <svg
                  color="var(--iw-color-arrowDown)"
                  fill="currentColor"
                  height="30"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex w-full p-2 rounded-lg border-2 border-iw-shieldLoadingShadow bg-iw-disclaimerBackGround my-8"
              data-testid="DataShareDisclaimer-Outer-Container"
            >
              <div
                class="px-4 text-center text-iw-disclaimerText text-xs"
                data-testid="DataShareDisclaimer-Content"
              >
                Please note that the QR Code can be shared only for the number of times you have chosen here. To use the QR Code for sharing again, you have to download the PDF again.
              </div>
            </div>
          </div>
          <div
            class="flex items-center justify-between p-6 rounded-b"
            data-testid="DataShareFooter-Outer-Container"
          >
            <div
              class="w-full"
            >
              <div
                class="bg-gradient-to-r rounded-lg text-center cursor-pointer shadow hover:shadow-lg from-iw-primary to-iw-secondary p-0.5"
              >
                <div
                  class="py-2 px-4 rounded-lg bg-white"
                >
                  <button
                    class="text-iw-tertiary font-bold rounded-lg"
                    data-testid="DataShareFooter-Cancel-Button"
                  >
                    <span
                      class="bg-gradient-to-r from-iw-primary to-iw-secondary bg-clip-text text-md font-bold text-transparent"
                    >
                      cancel
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="w-full"
            >
              <div
                class="w-full bg-iw-primary text-center cursor-pointer items-center text-iw-text ml-2 font-bold text-sm px-6 py-3 rounded-lg shadow hover:shadow-lg mr-1 bg-gradient-to-r from-iw-primary to-iw-secondary"
              >
                <button
                  class="text-white font-bold rounded-lg"
                  data-testid="DataShareFooter-Success-Button"
                >
                  success
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="opacity-25 fixed inset-0 z-30 bg-iw-backDrop"
    data-testid="ModalWrapper-BackDrop"
  />
</DocumentFragment>
`;
